src/decoders/base32_decoder.rs

```diff
@@ -0,0 +1,189 @@
+        checkers::{
+            athena::Athena,
+            checker_type::{Check, Checker},
+            CheckerTypes,
+        },
+        decoders::interface::{Crack, Decoder},
+    };
+
+    // helper for tests
+    fn get_athena_checker() -> CheckerTypes {
+        let athena_checker = Checker::<Athena>::new();
+        CheckerTypes::CheckAthena(athena_checker)
+    }
+
+    #[test]
+    fn successful_decoding() {
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+
+        let result = base32_decoder.crack("NBSWY3DPEB3W64TMMQ======", &get_athena_checker());
+        let decoded_str = &result
+            .unencrypted_text
+            .expect("No unencrypted text for base32");
+        assert_eq!(decoded_str, "hello world");
+    }
+
+    #[test]
+    fn base32_decode_empty_string() {
+        // Bsae64 returns an empty string, this is a valid base64 string
+        // but returns False on check_string_success
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+        let result = base32_decoder
+            .crack("", &get_athena_checker())
+            .unencrypted_text;
+        assert!(result.is_none());
+    }
+
+    #[test]
+    fn base32_decode_handles_panics() {
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+        let result = base32_decoder
+            .crack(
+                "hello my name is panicky mc panic face!",
+                &get_athena_checker(),
+            )
+            .unencrypted_text;
+        if result.is_some() {
+            panic!("Decode_base32 did not return an option with Some<t>.")
+        } else {
+            // If we get here, the test passed
+            // Because the base64_decoder.crack function returned None
+            // as it should do for the input
+            assert_eq!(true, true);
+        }
+    }
+
+    #[test]
+    fn base32_handle_panic_if_empty_string() {
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+        let result = base32_decoder
+            .crack("", &get_athena_checker())
+            .unencrypted_text;
+        if result.is_some() {
+            assert_eq!(true, true);
+        }
+    }
+
+    #[test]
+    fn base32_work_if_string_not_base32() {
+        // You can base64 decode a string that is not base64
+        // This string decodes to:
+        // ```.ée¢
+        // (uÖ²```
+        // https://gchq.github.io/CyberChef/#recipe=From_Base64('A-Za-z0-9%2B/%3D',true)&input=aGVsbG8gZ29vZCBkYXkh
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+        let result = base32_decoder
+            .crack("hello good day!", &get_athena_checker())
+            .unencrypted_text;
+        if result.is_some() {
+            assert_eq!(true, true);
+        }
+    }
+
+    #[test]
+    fn base32_handle_panic_if_emoji() {
+        let base32_decoder = Decoder::<Base32Decoder>::new();
+        let result = base32_decoder
+            .crack("😂", &get_athena_checker())
+            .unencrypted_text;
```

Comment:
Theoretically should we be able to decode an emoji with base32?

Reply:
No, it doesn't work.
https://gchq.github.io/CyberChef/#recipe=From_Base32('A-Z2-7%3D',false)&input=8J%2BYgg